tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -1231,6 +1231,70 @@ def get_kurtosis(y):
     return zip(index, res)
 
 
+@set_property("fctype", "combiner")
+@set_property("input", "pd.Series")
+@set_property("index_type", pd.DatetimeIndex)
+def energy_content_frequency_brackets(
+    signal: pd.Series, params: dict = {"number_of_bins": 10, "with_normalization": True}
```

Comment:
Please do not set the default parameter to a dictionary (https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python). This is not only related to `tsfresh` but for general python: this can lead to very unexpected behaviour (see the linked stackoverflow if you want to know more).
(you even disabled the "dangerous default parameter" pylint check for this ;) )

The good thing is, you do not even need a default argument. tsfresh will always fill this argument, so you can just remove the default.

Reply:
Thanks for adding type information. But `params` is actually a list of dictionaries.
As you chose a combiner calculator (https://tsfresh.readthedocs.io/en/latest/text/how_to_add_custom_feature.html#step-1-decide-which-type-of-feature-you-want-to-implement) you will get all parameters for this calculator as a list of dictionaries.

This will also need some changes to your code.

Reply:
From that point of view, I'm a bit unsure if I should leave the calculator to be a combiner or convert it to a "simple" calculator. Converting it into a "simple" calculator would allow me to simplify the code, and speed up calculation. For the time being I therefore decided to convert it, and would appreciate feedback if that decision is appropriate. 
Thanks!